contracts/interfaces/hats/IHats.sol

```diff
@@ -39,4 +39,15 @@ interface IHats {
     ) external returns (bool success);
 
     function transferHat(uint256 _hatId, address _from, address _to) external;
+
+    function getHatEligibilityModule(
+        uint256 _hatId
+    ) external view returns (address eligibility);
+
+    function isWearerOfHat(
+        address _user,
+        uint256 _hatId
+    ) external view returns (bool isWearer);
+
+    function changeHatEligibility(uint256 _hatId, address _newEligibility) external;
```

Comment:
Please see the following commits and discussions from other PRs:

- Relevant discussion, but for Sablier not Hats: https://github.com/decentdao/decent-contracts/pull/100#discussion_r1792349087

- Relevant commit which implements what I think should happen in this branch, too: https://github.com/decentdao/decent-contracts/pull/108/commits/e1fd7d75a9ba03aa9f47c4fef03a8cf3f359eace

Please let me know if it's not clear why we need to do this. The TL;DR is that if we _don't_, then you'll notice that running the deployment scripts will result in a new instance of `DecentHats_0_1_0` being deployed (because the bytecode has changed), and we need to avoid that.

Reply:
AH you know I watched for that and hadn't looked at the deployment diff yet ha. cool beans I'll take a look at what you've done and update according

Reply:
even better 🙏

Reply:
@Da-Colon holy shit it ended up being 16 commits LMAO

But that's only because I did things very atomically. Recommend you read through them in order to see what I did.